Tests/test_image_paste.py

```diff
@@ -101,226 +103,303 @@ def gradient_RGBa(self):
-        for mode in ("RGBA", "RGB", "L"):
-            im = Image.new(mode, (200, 200), (50, 60, 70, 80)[: len(mode)])
-            color = (10, 20, 30, 40)[: len(mode)]
-
-            self.assert_9points_paste(
-                im,
-                color,
-                self.mask_1,
-                [
-                    (50, 60, 70, 80),
-                    (50, 60, 70, 80),
-                    (10, 20, 30, 40),
-                    (50, 60, 70, 80),
-                    (50, 60, 70, 80),
-                    (10, 20, 30, 40),
-                    (10, 20, 30, 40),
-                    (10, 20, 30, 40),
-                    (50, 60, 70, 80),
-                ],
-            )
-
-    def test_color_mask_L(self):
-        for mode in ("RGBA", "RGB", "L"):
-            im = getattr(self, "gradient_" + mode).copy()
-            color = "white"
-
-            self.assert_9points_paste(
-                im,
-                color,
-                self.mask_L,
-                [
-                    (127, 191, 254, 191),
-                    (111, 207, 206, 110),
-                    (127, 254, 127, 0),
-                    (207, 207, 239, 239),
-                    (191, 191, 190, 191),
-                    (207, 206, 111, 112),
-                    (254, 254, 254, 255),
-                    (239, 206, 206, 238),
-                    (254, 191, 127, 191),
-                ],
-            )
-
-    def test_color_mask_RGBA(self):
-        for mode in ("RGBA", "RGB", "L"):
-            im = getattr(self, "gradient_" + mode).copy()
-            color = "white"
-
-            self.assert_9points_paste(
-                im,
-                color,
-                self.gradient_RGBA,
-                [
-                    (127, 191, 254, 191),
-                    (111, 207, 206, 110),
-                    (127, 254, 127, 0),
-                    (207, 207, 239, 239),
-                    (191, 191, 190, 191),
-                    (207, 206, 111, 112),
-                    (254, 254, 254, 255),
-                    (239, 206, 206, 238),
-                    (254, 191, 127, 191),
-                ],
-            )
-
-    def test_color_mask_RGBa(self):
-        for mode in ("RGBA", "RGB", "L"):
-            im = getattr(self, "gradient_" + mode).copy()
-            color = "white"
-
-            self.assert_9points_paste(
-                im,
-                color,
-                self.gradient_RGBa,
-                [
-                    (255, 63, 126, 63),
-                    (47, 143, 142, 46),
-                    (126, 253, 126, 255),
-                    (15, 15, 47, 47),
-                    (63, 63, 62, 63),
-                    (142, 141, 46, 47),
-                    (255, 255, 255, 0),
-                    (48, 15, 15, 47),
-                    (126, 63, 255, 63),
-                ],
-            )
+    @pytest.mark.parametrize(
+        "mode",
+        [
+            "RGBA",
+            "RGB",
+            "L",
+        ],
+    )
```

Comment:
This (and the others below) are fine as one-liners:
```suggestion
    @pytest.mark.parametrize("mode", ["RGBA", "RGB", "L"])
```